DTS-STN/Service-Canada-Labs
# TopNavBar Component

# [Create and add new top nav component to Layout.js](https://dev.azure.com/VP-BD/DECD/_boards/board/t/Service%20Canada%20Labs/Stories%20and%20Activities?workitem=215263)

This PR includes the new Nav bar that can be found in the new page designs as well as an accompanying story entry and unit test.

Figma:
https://www.figma.com/design/ke3SrZNGMP3AjwzMMuEgbo/%5Bcurrent%5D-BDM-DECD-SC-Labs?node-id=19176-10517&t=VRowRYItSVefILJe-4

Desktop:
![Screenshot 2024-06-04 at 3 39 59 PM](https://github.com/DTS-STN/Service-Canada-Labs/assets/31868510/ab12e696-3e9e-4d78-984a-cd9cad5ba5a5)

Mobile:
![Screenshot 2024-06-04 at 3 39 17 PM](https://github.com/DTS-STN/Service-Canada-Labs/assets/31868510/556e10c5-ee1b-4fc4-84bf-bed1bbdb0755)

By default, the TopNavBar is not active on the page. This is because it has links to pages that do not currently exist. The plan is to make it active when the new pages are ready. Nonetheless, I wanted to merge this branch as it includes some modifications to our default styles to `<ul>, <ol> and <li>` elements as well as some smaller tweaks.

Basically, list elements will by default have no styling so that we don't have to reverse default styles when using lists in certain contexts (like the nav menu in this PR). I've reviewed and made the appropriate changes to all pages where we use these elements, and applied the old default styles to the list elements in our TextNodeRender and FragmentRender so generated content should always be styled as we expect.

## Test Instructions

1. Go to Layout.js and set `isTopNavBarActive` to `true`
2. `yarn dev`
3. Navigate to home page or any other page
4. See that TopNavBar renders correctly, toggle language, and is desktop and mobile responsive
